lift code from docker/volume/mounts for splitting windows volumes

damoxc opened the pull request:
Using the API as provided from the `mounts` package imposes validation
on the `src:dest` which shouldn't be performed at this time. To workaround
that lift the internal code from that library required to only perform
the split.

Just wanted to share the patch I've used to fix Nomad for our use case, I'm not sure if this is the road you will want to down though.

A proposed fix for #5798.

hashicorp-cla commented:
[![CLA assistant check](https://cla.hashicorp.com/pull/badge/not_signed)](https://cla.hashicorp.com/hashicorp/nomad?pullRequest=5811)


Thank you for your submission! We require that all contributors sign our Contributor License Agreement ("CLA") before we can accept the contribution. [Read and sign the agreement](https://cla.hashicorp.com/hashicorp/nomad?pullRequest=5811)

[Learn more about why HashiCorp requires a CLA and what the CLA includes](https://www.hashicorp.com/cla)

<sub>Have you signed the CLA already but the status is still pending? [Recheck it](https://cla.hashicorp.com/check/hashicorp/nomad?pullRequest=5811).</sub>

notnoop commented:
Thanks @damoxc, this seems promising indeed !  I would suggest moving lifted code to a separate file and include a comment linking to original source along with license notice.

damoxc commented:
@notnoop My pleasure! I've moved the code out to a new file and added some comments to the top to the source file and license in the Docker repository, is that what you were after?

notnoop commented:
Thank you so much for your contribution!  I'll follow up and update appveyor tests (which sadly don't run on PR builds).
